### PR TITLE
Align management state constant with other constants

### PR DESCRIPTION
**Pull request by pavolloffay**

### CODE REVIEW

File: apis/v1alpha1/opentelemetrycollector_types.go [hunk 1]
```diff
@@ -23,17 +23,17 @@ import (
...
 //
-// +kubebuilder:validation:Enum=Managed;Unmanaged
+// +kubebuilder:validation:Enum=managed;unmanaged
 type ManagementStateType string
 
 const (
 	// ManagementStateManaged when the OpenTelemetryCollector custom resource should be
 	// reconciled by the operator.
-	ManagementStateManaged ManagementStateType = "Managed"
+	ManagementStateManaged ManagementStateType = "managed"
```
Comment (pavolloffay) [inline apis/v1alpha1/opentelemetrycollector_types.go:32]: We don't use constants with capital letters in the codebase.

**Comment by swiatekm-sumo** [comment]
Isn't this a breaking change we should bump the apiVersion for?

**Comment by iblancasa** [comment]
> Isn't this a breaking change we should bump the apiVersion for?

It was added a couple of days ago #1888. It should not be needed, AFAIK

File: docs/api.md [hunk 2]
```diff
@@ -3708,7 +3708,7 @@ OpenTelemetryCollectorSpec defines the desired state of OpenTelemetryCollector.
         <td>
           ManagementState defines if the CR should be managed by the operator or not. Default is managed.<br/>
           <br/>
-            <i>Enum</i>: Managed, Unmanaged<br/>
+            <i>Enum</i>: managed, unmanaged<br/>
             <i>Default</i>: Managed<br/>
```
Comment (iblancasa) [inline docs/api.md:3712]:
```suggestion
            <i>Default</i>: managed<br/>
```

**Review by iblancasa — COMMENTED** [review]
The E2E need to be updated too

**Comment by pavolloffay** [comment]
> Isn't this a breaking change we should bump the apiVersion for?

It's not, the feature was not released

**Review by jaronoff97 — APPROVED** [review]

**Pull request merged**